FluxNotes/flux
# Link clinical notes and encounters

Addresses JIRA-1963.

-  Added `resultingClinicalNote` property to `FluxConsultRequested` which is a reference to a `FluxClinicalNote` entry.
-  Added `documentedEncounter` property to `FluxClinicalNote` which is a reference to a `FluxConsultRequested` entry.
-  Updated `GistAdjuvantIhanos.json` with `resultingClinicalNote` and `documentedEncounter` properties.
-  Updated dates of encounters in Ihanos to match with clinical notes.
-  When adding a new note a linkage is created between the closest encounter(determined by absolute time) and the note.  Will remove linkage from the encounter when a note is deleted.

Things to test/look for:
-  Make sure dates and linkages in Ihanos are correct.  I've double checked the properties added but easily could've made an error.  If an encounter has a link to a note, then that note should have a link to that encounter.
-  Create a new note and and view the list of entries to verify a linkage was created to an encounter.
    -  Delete the note and make sure linkage is removed in the encounter.
-  Create additional notes and verify that no linkages are added on the note entry since there shouldn't exist any unlinked encounters.

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._


## Submitter:

**Running the Application**


**Task**


**Documentation**

-  Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
-  Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
-  Document any new APIs/extension points
-  Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**


**Tests**

-  Added backend tests for new functionality added


## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code
